backup-commons/src/main/java/com/opsgenie/tools/backup/util/BackupUtils.java

```diff
@@ -46,6 +47,20 @@ public static String readFile(String fileName) throws IOException {
 
         return sb.toString();
     }
+    public static String getTeamNameFromId(File teamsDirectory, String teamId) {
+        String[] files = getFileListOf(teamsDirectory);
+        if (files == null || files.length == 0) {
+            logger.warn("Warning: {} is empty", teamsDirectory);
+            return null;
```

Comment:
Will it be possible to have Optional<String> rather than String which would decrease the chance of NPE? If it is not a large task.

Reply:
Since the current Java version for this project is set as 1.6, we won't be able to use any Java 8 features. Hence can't really use `Optional` without updating the java version.

Reply:
can we return an empty array here instead of returning null?

Reply:
Can we create out own Optional implementation something similar to : 

```java
public class Optional<T> {
    private final T value;

    private Optional(T value) {
        this.value = value;
    }

    public static <T> Optional<T> of(T value) {
        if (value == null) {
            throw new NullPointerException("Value cannot be null");
        }
        return new Optional<T>(value);
    }

    public static <T> Optional<T> ofNullable(T value) {
        return new Optional<T>(value);
    }

    public T get() {
        if (value == null) {
            throw new IllegalStateException("Value is not present");
        }
        return value;
    }

    public boolean isPresent() {
        return value != null;
    }

    public T orElse(T defaultValue) {
        return value != null ? value : defaultValue;
    }
}
```       

Usage:

```java
public class Main {
    public static void main(String[] args) {
        Optional<String> stringOptional = Optional.ofNullable(getStringValue());

        if (stringOptional.isPresent()) {
            System.out.println("Value: " + stringOptional.get());
        } else {
            System.out.println("Value not present");
        }
    }

    private static String getStringValue() {
        // Your logic to return a string or null
        return "Hello, World!";
    }
}
```

Reply:
Added a custom  `OptionalUtil`